packages/common/src/getContract.ts

```diff
@@ -79,16 +79,16 @@ export function getContract<
   TPublicClient,
   TWalletClient
 >): GetContractReturnType<TAbi, { public: TPublicClient; wallet: TWalletClient }, TAddress> {
-  const contract = viem_getContract({
+  const contract: GetContractReturnType<TAbi, { public: TPublicClient; wallet: TWalletClient }, TAddress> & {
+    write: unknown;
+  } = viem_getContract({
     abi,
     address,
     client: {
       public: publicClient,
       wallet: walletClient,
     },
-  }) as unknown as GetContractReturnType<TAbi, { public: TPublicClient; wallet: TWalletClient }, TAddress> & {
-    write: unknown;
-  };
+  }) as never;
```

Comment:
is there any practical benefit to this approach?

Reply:
Yes, a few things:

- Avoid duplicating long, complex expressions with no additional type safety
- Avoid comparing potentially complex expressions that don't need to be compared because you're casting them anyways. This comparison shouldn't occur in the `as unknown as X` case but it would occur if e.g. you have an explicit return type and also case the return value in some areas to another complex expression instead of just `never` when the only goal is just to allow that return anyways.

Reply:
happy with this change, but just FYI, I reverted this back to what's in main and I see no change in instantiations

Reply:
Yeah I should have been clearer, I made this change while optimizing perf based on the traces I saw but it just led me to the variance issue in viem. I know there is no perfect impact here, and specifically casting `as unknown as X` for a function with an explicit return type of `X` will not cost anything.

The problem is if `X` is a long expression, not only is it unnecessary duplication, if the two ever get slightly out of sync, e.g. an extra optional parameter is added to the return but not the cast, then TS does have to compare the instantiations which can be very expensive.